pallets/phala/src/compute/base_pool.rs

```diff
@@ -378,14 +370,18 @@ pub mod pallet {
 				// The share held by the vault
 				let mut vault = ensure_vault::<T>(*vault_staker)
 					.expect("vault in value_subscribers should always exist: qed.");
-				let nft_id = Pallet::<T>::merge_or_init_nft_for_staker(
+				let maybe_nft_id = Pallet::<T>::merge_nft_for_staker(
 					self.cid,
 					vault.basepool.pool_account_id.clone(),
 					self.pid,
-					PoolType::StakePool,
 				)
 				.expect("merge nft shoule always success: qed.");
 
+				if maybe_nft_id.is_none() {
+					// Never get here
+					continue;
+				}
+				let nft_id = maybe_nft_id.unwrap();
```

Comment:
```suggestion
				let Some(nft_id) = maybe_nft_id else {
					// Never get here
					continue;
				};
```

`let-else` syntax can make the logic clearer.